src/snmalloc/backend_helpers/pagemap.h

```diff
@@ -117,31 +117,43 @@ namespace snmalloc
 #endif
       SNMALLOC_ASSERT(s != 0);
       // TODO take account of pagemap size in the calculation of how big it
-      // needs to be.
+      // needs to be.  The following code creates a pagemap that covers the
+      // pagemap as well as the left over. This is not ideal, and we should
+      // really calculate the division with
+      //
+      //  bits::one_at_bit(GRANULARITY_BITS) + sizeof(T)
+      //
+      // There are awkward corner cases for the alignment of the start and
+      // the end that are hard to calculate. So this is not currently done.
 
-      // Align the start and end.  We won't store for the very ends as they
-      // are not aligned to a chunk boundary.
-      auto heap_base = pointer_align_up(b, bits::one_at_bit(GRANULARITY_BITS));
-      auto end = pointer_align_down(
+      // Calculate range in pagemap that is associated to this space.
+      // Over calculate to cover any unaligned parts at either end.
+      auto b_align = pointer_align_down(b, bits::one_at_bit(GRANULARITY_BITS));
+      auto end = pointer_align_up(
```

Comment:
Similarly end only seems to be used as an address to we can avoid the potentially out of bounds pointer floating around.